packer/provision/docker.sh

```diff
@@ -7,9 +7,25 @@ set -xeu -o pipefail
 
 rh_changes() {
     echo "---> RH changes"
+    # Following directions from 
+    # https://docs.docker.com/engine/installation/linux/docker-ce/centos/
+
+    # remove old docker
+    yum remove -y docker docker-common docker-selinux docker-engine
+
+    # set up the repository
+    yum install -y yum-utils device-mapper-persistent-data lvm2
+    yum-config-manager --add-repo \
+    https://download.docker.com/linux/centos/docker-ce.repo
+
+    # update package repo
+    # figure out how to gpg verify the next step
+    # or can I just do yum -y update?
+
+    #####   sudo yum makecache fast
```

Comment:
If you force clear the metadata cache (yum clean metadata) and then follow that with the yum install -y it should take care of the GPG key IIRC.

Otherwise you will need to do the following:

rpm --import $GPG_KEY_URL

$GPG_KEY_URL will be in the repo file

Reply:
Yep that works.